cpp/src/io/orc/stripe_data.cu

```diff
@@ -1211,8 +1211,9 @@ __global__ void __launch_bounds__(block_size)
         uint32_t skippedrows = min(static_cast<uint32_t>(first_row - row_in), nrows);
         uint32_t skip_count  = 0;
         for (uint32_t i = t * 32; i < skippedrows; i += 32 * 32) {
-          uint32_t bits = s->vals.u32[i >> 5];
-          if (i + 32 > skippedrows) { bits &= (1 << (skippedrows - i)) - 1; }
```

Comment:
The issue is that the mask here selected the last `skippedrows - i` bits, instead of first?

Reply:
Yes, because of that if it misses any null in the first set of bits, then skipcount will be more and it will skip more than expected values.

Reply:
would this select the right bits?
```
auto const num_extra_bits = 32 - i + skippedrows;
bits &= ~((1 << num_extra_bits) - 1);
```